plugins/vpp/ipsecplugin/vppcalls/dump_vppcalls.go

```diff
@@ -0,0 +1,385 @@
+}
+
+func (handler *ipSecVppHandler) DumpIPSecSAWithIndex(saID uint32) (saList []*IPSecSaDetails, err error) {
+	defer func(t time.Time) {
+		handler.stopwatch.TimeLog(ipsecapi.IpsecSaDump{}).LogTimeEntry(time.Since(t))
+	}(time.Now())
+
+	saDetails, err := handler.dumpSecurityAssociations(saID)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, saData := range saDetails {
+		// Skip tunnel interfaces
+		if saData.SwIfIndex != ^uint32(0) {
+			continue
+		}
+
+		// Addresses
+		var tunnelSrcAddrStr, tunnelDstAddrStr string
+		if uintToBool(saData.IsTunnelIP6) {
+			var tunnelSrcAddr, tunnelDstAddr net.IP = saData.TunnelSrcAddr, saData.TunnelDstAddr
+			tunnelSrcAddrStr, tunnelDstAddrStr = tunnelSrcAddr.String(), tunnelDstAddr.String()
+		} else {
+			var tunnelSrcAddr, tunnelDstAddr net.IP = saData.TunnelSrcAddr[:4], saData.TunnelDstAddr[:4]
+			tunnelSrcAddrStr, tunnelDstAddrStr = tunnelSrcAddr.String(), tunnelDstAddr.String()
+		}
+
+		sa := &ipsec.SecurityAssociations_SA{
+			Spi:            saData.Spi,
+			Protocol:       getSaProto(saData.Protocol),
+			CryptoAlg:      getCryptoAlg(saData.CryptoAlg),
+			CryptoKey:      string(bytes.SplitN(saData.CryptoKey, []byte{0x00}, 2)[0]),
+			IntegAlg:       getIntegAlg(saData.IntegAlg),
+			IntegKey:       string(bytes.SplitN(saData.IntegKey, []byte{0x00}, 2)[0]),
+			UseEsn:         uintToBool(saData.UseEsn),
+			UseAntiReplay:  uintToBool(saData.UseAntiReplay),
+			TunnelSrcAddr:  tunnelSrcAddrStr,
+			TunnelDstAddr:  tunnelDstAddrStr,
+			EnableUdpEncap: uintToBool(saData.UDPEncap),
+		}
+		meta := &IPSecSaMeta{
+			SaID:           saData.SaID,
+			IfIdx:          saData.SwIfIndex,
+			CryptoKeyLen:   saData.CryptoKeyLen,
+			IntegKeyLen:    saData.IntegKeyLen,
+			Salt:           saData.Salt,
+			SeqOutbound:    saData.SeqOutbound,
+			LastSeqInbound: saData.LastSeqInbound,
+			ReplayWindow:   saData.ReplayWindow,
+			TotalDataSize:  saData.TotalDataSize,
+		}
+		saList = append(saList, &IPSecSaDetails{
+			Sa:   sa,
+			Meta: meta,
+		})
+	}
+
+	return saList, nil
+}
+
+// IPSecTunnelInterfaceDetails hold a list of tunnel interfaces with name/index map as metadata
+type IPSecTunnelInterfaceDetails struct {
+	Tunnels []*ipsec.TunnelInterfaces_Tunnel
+	Meta    *IPSecTunnelMeta
+}
+
+// IPSecTunnelMeta contains map of name/index pairs
+type IPSecTunnelMeta struct {
+	IfNameToIdx map[uint32]string
+}
+
+func (handler *ipSecVppHandler) DumpIPSecTunnelInterfaces() (tun *IPSecTunnelInterfaceDetails, err error) {
+	defer func(t time.Time) {
+		handler.stopwatch.TimeLog(ipsecapi.IpsecSaDump{}).LogTimeEntry(time.Since(t))
+	}(time.Now())
+
+	var tunnels []*ipsec.TunnelInterfaces_Tunnel
+	meta := &IPSecTunnelMeta{
+		IfNameToIdx: make(map[uint32]string),
+	}
+	saDetails, err := handler.dumpSecurityAssociations(^uint32(0))
+	if err != nil {
+		return nil, err
+	}
+
+	for _, saData := range saDetails {
+		// Skip non-tunnel security associations
+		if saData.SwIfIndex == ^uint32(0) {
```

Comment:
It seems that this constant value is being so many time we could actually declare it?